task/runner.go

```diff
@@ -299,6 +299,9 @@ func (r *runner) handleAMQPMessage(msg amqp.Delivery) (err error) {
 func (r *runner) handleTask(ctx context.Context, taskInfo data.TaskInfo) (out *TaskHandlerOutput, err error) {
 	taskCtx, err := r.buildTaskContext(ctx, taskInfo)
 	if err != nil {
+		if errors.Is(err, assetNotFound) {
+			return nil, UnretriableError{errors.New("task cancelled, asset not found")}
```

Comment:
It's a bit tricky to pass the assetID back up to this function but I can put the taskID in and then you could easily look up the assetID for the task when debugging

Reply:
`taskCtx.InputAssetID`, `taskCtx.OutputAssetID` should be available in this line

Reply:
Actually, this is the user facing error that will be set on the task object, so it is redundant to have the task ID in it. Since it only exists in the context of a specific task, the user will already have the task in hand, from which it can also obtain the asset ID. 

So in order to have consistent error messages also easily aggregatable in our reports, I'd suggest to keep this error as a fixed string as before.

Reply:
Perhaps the task/asset IDs would be more useful in an internal log insetead, not a user facing error message. Perhaps that would be better here? 

i.e. clean-up the user-facing error message and add a `glog.Warningf` log here containing any internal data we want to assist on our own debugging efforts.

WDYT?

Reply:
unfortunately taskCtx is going to be nil when an error is returned. But that makes sense to change it to a fixed string and add a log line :+1